client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/index.tsx

```diff
@@ -64,6 +67,7 @@ const SiteMigrationCredentials: Step< {
 		applicationPasswordsInfo?: ApplicationPasswordsInfo
 	) => {
 		const action = getAction( siteInfo, applicationPasswordsInfo );
+		dispatch( resetSite( siteId ) );
```

Comment:
As above, I think we need to check that we have a siteId first.
```suggestion
		siteId && dispatch( resetSite( siteId ) );
```